### ISSUE TITLE
java.lang.IndexOutOfBoundsException

### ISSUE THREAD

**Issue by hbzhzw** [square/okhttp#5029]
this bug always exists!   but can not repeat it, maybe can add protecting code to avoid exception ...

java.lang.IndexOutOfBoundsException: Index: 0
java.util.Collections$EmptyList.get(Collections.java:4502)
at okhttp3.internal.connection.RealConnection.connectTls(SourceFile:326)
at okhttp3.internal.connection.RealConnection.establishProtocol(SourceFile:283)
at okhttp3.internal.connection.RealConnection.connect(SourceFile:168)
at okhttp3.internal.connection.StreamAllocation.findConnection(SourceFile:257)
at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(SourceFile:135)
at okhttp3.internal.connection.StreamAllocation.newStream(SourceFile:114)
at okhttp3.internal.connection.ConnectInterceptor.intercept(SourceFile:42)
at okhttp3.internal.http.RealInterceptorChain.proceed(SourceFile:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(SourceFile:121)
at okhttp3.internal.cache.CacheInterceptor.intercept(SourceFile:93)
at okhttp3.internal.http.RealInterceptorChain.proceed(SourceFile:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(SourceFile:121)
at okhttp3.internal.http.BridgeInterceptor.intercept(SourceFile:93)
at okhttp3.internal.http.RealInterceptorChain.proceed(SourceFile:147)
at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(SourceFile:126)
at okhttp3.internal.http.RealInterceptorChain.proceed(SourceFile:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(SourceFile:121)
at okhttp3.RealCall.getResponseWithInterceptorChain(SourceFile:254)
at okhttp3.RealCall$AsyncCall.execute(SourceFile:200)
at okhttp3.internal.NamedRunnable.run(SourceFile:32)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:784)

**Comment by TangHuaiZhe** [comment]
Is your okhttp version too old?
The latest version has been protected:
`RealConnection.connectTls()`
```
        if (!peerCertificates.isEmpty()) {
          X509Certificate cert = (X509Certificate) peerCertificates.get(0);
```

**Comment by MonikaJethani** [comment]
@swankjesse @JakeWharton Can I work on this?

**Comment by rivu007** [comment]
Is there any update on this?
In our company we are facing the same issue. Can't publish report to Sonar server.

Using java 11 with grdle 5.4.1

Is there any workaround at the moment?

**Comment by swankjesse** [comment]
Executable test case please.

**Comment by TangHuaiZhe** [comment]
Same with  [IndexOutOfBoundsException in RealConnection #4427](https://github.com/square/okhttp/issues/4427)?
It should be fixed in 3.12.2 and later version.

**Comment by rivu007** [comment]
NVM, got it fixed. the issue was related to oracle jdk 11.0.2 : https://stackoverflow.com/questions/52016415/jdk-11-ssl-error-on-valid-certificate-working-in-previous-versions

Upgraded it to 11.0.3 and got it working.
